[hm-yang1] iP

hm-yang1 opened the pull request:
# CheeseBot
>"Life is great. Cheese makes it better." -- Anonymous

CheeseBot frees your mind of having to remember things you need to do. It's:
- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is:
- download it from [here](https://github.com/hm-yang1/ip).
- double-click it.
- add your tasks.
- let it manage your tasks for you 😄 

And it is **FREE!**

Features:
- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders
---
If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public class WheelyBigCheese {
    public static void main(String[] args) {
        WheelyBigCheese bot = new WheelyBigCheese();
        bot.run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

hm-yang1 committed:
Level 0. Rename, Greet, Exit
Deleted Duke.java, added WheelyBigCheese.java

hm-yang1 committed:
Level 1. Echo. Added ability for chat bot to echo user input

hm-yang1 committed:
Level 2. Add, List

hm-yang1 committed:
Level 3. Mark as Done

hm-yang1 committed:
Level 4. ToDos, Events, Deadlines

hm-yang1 committed:
A-TextUiTesting

hm-yang1 committed:
Level 5. Handle Errors

hm-yang1 committed:
Level 6. Delete

hm-yang1 committed:
Level-7: Save

* add save task list to txt file
* arraylist for task moved for better encapsulation

hm-yang1 committed:
Add directory where tasklist saved

hm-yang1 committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add directory where tasklist saved
  Level-7: Save

hm-yang1 committed:
Level 8. Dates and Times

* change date to LocalDate, for better use

hm-yang1 committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Level 8. Dates and Times

hm-yang1 committed:
Add more oop

*Ui: deals with interactions with the user
*Storage: deals with loading tasks from the file and saving tasks in the file
*Parser: deals with making sense of the user command
*TaskList: contains the task list e.g., it has operations to add/delete tasks in the list
*Command: contains logic for various operations of bot

hm-yang1 committed:
Merge branch 'branch-more-oop'

* branch-more-oop:
  Add more oop

hm-yang1 committed:
A-Packages

*Divide classes into packages

hm-yang1 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

* origin/add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

hm-yang1 committed:
A-Gradle

hm-yang1 committed:
A-JUnit

hm-yang1 committed:
A-Jar

hm-yang1 committed:
A-JavaDoc

More JavaDoc comments

hm-yang1 committed:
Level-9

Add substring search for tasks

hm-yang1 committed:
A-CodingStandard

*Add checkstyle
*Fix most style errors

hm-yang1 committed:
Merge branch 'branch-coding-standard'

* branch-coding-standard:
  A-CodingStandard

hm-yang1 committed:
Merge branch 'branch-level-9'

* branch-level-9:
  Level-9

hm-yang1 committed:
CheckStyle improvements

hm-yang1 committed:
A-JavaDoc correction

*Wrong branch name for A-JavaDoc
*Add update to correct branch name

hm-yang1 committed:
A-CodingStandard correction

*Again, wrong branch name for A-CodingStandard
*Add commit to correct

hm-yang1 committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  A-JavaDoc correction

hm-yang1 committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  A-CodingStandard correction

hm-yang1 committed:
Add A-CheckStyle

hm-yang1 committed:
Merge tag 'A-CheckStyle'

* tag 'A-CheckStyle':
  Add A-CheckStyle

hm-yang1 committed:
Improve A-CheckStyle

hm-yang1 committed:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  Improve A-CheckStyle

hm-yang1 committed:
Partial level 10 GUI

Update bot to work with GUI. Commands and UI now return String responses in addition to printing out the responses

hm-yang1 committed:
Level 10 GUI

Complete GUI for chatbot, very similar to javafx tutorial

hm-yang1 committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Level 10 GUI
  Partial level 10 GUI

myrilla12 commented on src/main/java/cheese/CheeseException.java:
```diff
@@ -0,0 +1,14 @@
+package cheese;
+
+/**
+ * Custom exception
```
JavaDoc header comments should start in the form of something like 'Customizes...' or 'Represents...'

myrilla12 commented on src/main/java/cheese/CheeseException.java:
```diff
@@ -0,0 +1,14 @@
+ */
+public class CheeseException extends Exception {
+    /**
+     * Just a constructor
```
This is a very minor comment but I think there should be a blank space in between the descriptor and any @param or @return. Each line should also be concluded with a full stop.

myrilla12 commented on src/main/java/cheese/Parser.java:
```diff
@@ -0,0 +1,81 @@
+     * @return int
+     * @throws CheeseException Missing/Incorrect input
+     */
+    public static int getIdx(String[] inputTokens, int size) throws CheeseException {
```
Could idx be a potentially confusing name? Perhaps consider extending it to index.

myrilla12 commented on src/main/java/cheese/Storage.java:
```diff
@@ -0,0 +1,106 @@
+     * Add and save new task
+     * @param t task to be added
+     */
+    public void add(Task t) throws CheeseException {
```
Could the naming appear a little inconsistent? The previous method was called loadTask while the following are just add and update rather than addTask and updateTask

myrilla12 commented on src/main/java/cheese/TaskList.java:
```diff
@@ -0,0 +1,36 @@
+        tasks.add(task);
+    }
+
+    public int size() {
```
Shouldn't the method name be a verb and not a noun? For example, getSize instead of size

myrilla12 commented on src/main/java/cheese/TaskList.java:
```diff
@@ -0,0 +1,36 @@
+        return tasks.size();
+    }
+
+    public Task get(int i) {
```
I personally find 'get' to be a vague method name

myrilla12 commented on src/main/java/cheese/command/AddCommand.java:
```diff
@@ -0,0 +1,37 @@
+ * Command to add tasks to list
+ */
+public class AddCommand extends Command {
+    private final Task task;
```
Shouldn't final variables be named in SCREAMING_SNAKE_CASE?

myrilla12 commented on src/main/java/cheese/command/Command.java:
```diff
@@ -0,0 +1,36 @@
+ * Encapsulates small parts of logic for the bot
+ */
+public class Command {
+    private final boolean exitChat;
```
Same with this final variable

ty4g1 replied:
To follow the coding standard, perhaps it would be better to rename this boolean to `isExitChat` or `isExitCommand` instead.

myrilla12 reviewed (commented):
LGTM. I think the JavaDoc comments might need a look over as they don't follow comment conventions. However, the code mostly follows convention aside from a few naming conventions.

ty4g1 commented on src/main/java/cheese/command/MarkCommand.java:
```diff
@@ -0,0 +1,40 @@
+ * Command to mark a task as done
+ */
+public class MarkCommand extends UpdateCommand {
+    private final boolean done;
```
Similar comment, would be better to rename it to `isDone` to follow the coding standard.

ty4g1 reviewed (commented):
LGTM, just a few nits with variable names to follow coding standard. I agree with the other comments, the JavaDocs could be more descriptive.

hm-yang1 committed:
Add assertions

hm-yang1 committed:
Improve code quality

* Add and improve comments
* Refactored code, especially in the Parser. Move the parsing in the tasks to the Parser
* Rename functions and variables to be more consistent

hm-yang1 committed:
Use Java Streams

* Use Java Streams to do searching
* Also use Java Lambdas
* Will try to find other places to use Java Streams. Cannot think of any right now.

hm-yang1 committed:
Merge pull request #1 from hm-yang1/branch-A-Assertions

Add assertions

hm-yang1 committed:
Merge branch 'master' of https://github.com/hm-yang1/ip

* 'master' of https://github.com/hm-yang1/ip:
  Add assertions

hm-yang1 committed:
Merge pull request #2 from hm-yang1/branch-A-CodeQuality

Improve code quality

hm-yang1 committed:
Merge branch 'master' into branch-A-Streams

* master:
  Improve code quality
  Add assertions

# Conflicts:
#	src/main/java/cheese/command/FindCommand.java

hm-yang1 committed:
Merge pull request #3 from hm-yang1/branch-A-Streams

Add Java Streams

hm-yang1 committed:
Merge branch 'master' of https://github.com/hm-yang1/ip

* 'master' of https://github.com/hm-yang1/ip:
  Use Java Streams

hm-yang1 committed:
B-Snooze

* Add way to snooze/reschedule tasks

hm-yang1 committed:
Merge pull request #4 from hm-yang1/branch-B-Snooze

B-Snooze

hm-yang1 committed:
Merge branch 'master' of https://github.com/hm-yang1/ip

* 'master' of https://github.com/hm-yang1/ip:
  B-Snooze

hm-yang1 committed:
Better GUI

* Add colored text boxes
* Improve profile pictures
* Improve resizing of images

hm-yang1 committed:
More testing

* Add testing for Parser, Storage, Ui and TaskList

hm-yang1 committed:
Style improvements

hm-yang1 committed:
Merge branch 'branch-A-BetterGui'

* branch-A-BetterGui:
  Better GUI

hm-yang1 committed:
Merge branch 'branch-A-MoreTesting'

* branch-A-MoreTesting:
  Style improvements
  More testing

hm-yang1 committed:
Add user guide

-Add user guide in Markdown
-Add Ui.png and other screenshots
-Change name of bot to be simpler

hm-yang1 committed:
Change name of bot class

hm-yang1 committed:
Fix spelling mistake
